### DISCUSSION TITLE
Sent attachments is not showing up

### DISCUSSION THREAD

**Discussion by ienam** [flyerhq/flutter_chat_ui#111] [Q&A] [answered]
hello. before i ask away, great flutter chat package btw. keep it up! =)

so here's my problem/inquiry, i'm creating a chat feature which uses chatlio methods/api's. i already have the basic text messaging worked. now my problem is with the attachments, sending pictures and files works fine (i think) in the app (because i can see it with the help of '_handleMessageTap' and '_handlePreviewDataFetched' methods. the problem is when it has been sent to slack it shows as plain text. how do i make it work so that when an attachment has been sent to slack it will still preview or work as expected (like if it's a file or an image)?

thanks so much in advance. appreciate any help! =)

**Comment by demchenkoalex** [comment, marked as answer]
Hi @ienam thank you!

Honestly, I don't know how I can help. The package was designed in a way so it can be used with whatever, that's why you have all those callbacks for sending images, files etc. so you can write your own implementation. ATM I was only used this chat UI with Firebase and I used Cloud Storage to upload files and images, and, after upload, I attached generated URL to the `ImageMessage` or `FileMessage` so it is rendering in the UI. See example [here](https://github.com/flyerhq/flutter_firebase_chat_core/blob/main/example/lib/chat.dart).

In your case either your service has some storage or you use AWS or something, you pass the URL to our library attaching it to the `File` or `Image` message, so it is rendered correctly here and Slack should be smart to parse the URL and see whether it is image or file, showing preview, download etc. That is how I think this should work, but I don't know the service and all the details, so these are just my thoughts.

**Reply by ienam** [reply]
thanks @demchenkoalex. i'll give it a try. =)
